lib/Conversion/TorchToLinalg/DataMovement.cpp

```diff
@@ -1834,6 +1834,147 @@ class ConvertAtenViewAsRealOp : public OpConversionPattern<AtenViewAsRealOp> {
+        loc, 
+        rewriter.create<arith::MinSIOp>(
+          loc, 
+          rewriter.create<arith::AddIOp>(
+            loc, 
+            dim1Size, 
+            rewriter.create<arith::ConstantOp>(loc, rewriter.getI64IntegerAttr(offset))
+          ), 
+          dim2Size
+        ), 
+        rewriter.create<arith::ConstantOp>(loc, rewriter.getI64IntegerAttr(0))
+      );
+    else
+      diagSize = rewriter.create<arith::MaxSIOp>(
+          loc, 
+          rewriter.create<arith::MinSIOp>(loc, rewriter.create<arith::SubIOp>(
+              loc,
+              dim2Size, 
+              rewriter.create<arith::ConstantOp>(loc, rewriter.getI64IntegerAttr(offset))
+          ), 
+          dim1Size), 
+          rewriter.create<arith::ConstantOp>(loc, rewriter.getI64IntegerAttr(0))
+      );
+    
+    // TODO: below version also works, need to cast diagSize to Value below
+    // int64_t diagSize;
+    // if (offsetNegative)
+    //   diagSize = std::max(std::min(dim1Size.cast<int64_t>() + offset, dim2Size.cast<int64_t>()), (int64_t)0);
+    // else
+    //   diagSize = std::max(std::min(dim2Size.cast<int64_t>() - offset, dim1Size.cast<int64_t>()), (int64_t)0);
+
+    int64_t diagStart1 = offsetNegative ? - offset : 0;
+    int64_t diagStart2 = offsetNegative ? 0 : offset;
+    
+    SmallVector<Value> outputDims;
+    for (auto i = 0; i < inputRank; i++)
+      if (!(i == dim1) && !(i == dim2))
+        outputDims.push_back(getDimOp(rewriter, loc, adaptor.getSelf(), i));
+    outputDims.push_back(diagSize); // TODO: use this or line below depending on which implementation we use above
+    // outputDims.push_back(rewriter.create<arith::ConstantOp>(loc, rewriter.getI64IntegerAttr(diagSize)));
+    
+    Value outMatrix = rewriter.create<tensor::EmptyOp>(
+        loc, getAsOpFoldResult(outputDims), elementType);
+    
+    SmallVector<AffineExpr> inExprs;
+    SmallVector<AffineExpr> diagExprs;
+    // Input indexing (i1, ... , in, j1, j2) -> (i1, ... , in, j1+off1, j1+off2) (also ranges have to change with this, is this done automatically?)
+    // Output indexing (i1, ... , in, j1, j2) -> (i1, ... , in, j1), i.e. output indexing is just reduction of last dim
+    for (auto i = 0; i < inputRank; i++)
+      if (i==dim1)
+        // TODO: limits will change here, just from 0 to diagSize, pass this somehow?
+        inExprs.push_back(diagStart1+getAffineDimExpr(dim1, rewriter.getContext())); // used operator overload instead of add constant function for affine dim expr
+      else if (i==dim2)
+        // TODO: limits will change here, just from 0 to diagSize, pass this somehow?
+        inExprs.push_back(diagStart2+getAffineDimExpr(dim1, rewriter.getContext())); // used operator overload
+      else
+        inExprs.push_back(getAffineDimExpr(i, rewriter.getContext())); // iterate over unchanged dims
+    
+    // for diagonal result, one of the dims is deleted and one is moved to the end
```

Comment:
Use upper case first letter and period at the end as in your other comments above?

Reply:
Thanks, that is a good remark. The comment does not occur anymore in the recent commits